Cargo build issue: failed to run custom build command for `openssl v0.7.14`

Hi,

New to Rust so there may be something obvious for me to fix here- but would appreciate some help.  

Fresh install of Rust, on Ubuntu w/ libssl-dev (1.1.0g-2ubuntu4.3) installed.

the build process will not complete due to an openssl dependency problem.   I've tried explicitly setting OPENSSL_INCLUDE_DIR and OPENSSL_INCLUDE_LIB env vars but still the same.

```
cargo build --release
   Compiling libc v0.2.44
   Compiling pkg-config v0.3.14
   Compiling num-traits v0.2.6
   Compiling proc-macro2 v0.4.24
   Compiling gcc v0.3.55
   Compiling num-integer v0.1.39
   Compiling num-bigint v0.2.1
   Compiling typenum v1.10.0
   Compiling num-complex v0.2.1
   Compiling unicode-xid v0.1.0
   Compiling cfg-if v0.1.6
   Compiling matches v0.1.8
   Compiling rand_core v0.3.0
   Compiling matrixmultiply v0.1.15
   Compiling cc v1.0.25
   Compiling num-iter v0.1.37
   Compiling unicode-normalization v0.1.7
   Compiling version_check v0.1.5
   Compiling winapi-build v0.1.1
   Compiling num-rational v0.2.1
   Compiling unicode-xid v0.0.4
   Compiling cgmath v0.16.1
   Compiling rustc-serialize v0.3.24
   Compiling serde v1.0.80
   Compiling siphasher v0.2.3
   Compiling httparse v1.3.3
   Compiling percent-encoding v1.0.1
   Compiling build_const v0.2.1
   Compiling nodrop v0.1.13
   Compiling libm v0.1.2
   Compiling void v1.0.2
   Compiling bitflags v0.7.0
   Compiling quote v0.3.15
   Compiling traitobject v0.1.0
   Compiling rawpointer v0.1.0
   Compiling lazy_static v0.2.11
   Compiling scopeguard v0.3.3
   Compiling semver v0.1.20
   Compiling procedural-masquerade v0.1.5
   Compiling slog v2.4.1
   Compiling protobuf v1.7.4
   Compiling failure_derive v0.1.3
   Compiling memoffset v0.2.1
   Compiling winapi v0.2.8
   Compiling futures v0.1.25
   Compiling lazy_static v1.2.0
   Compiling approx v0.1.1
   Compiling ryu v0.2.7
   Compiling adler32 v1.0.3
   Compiling encoding_rs v0.8.13
   Compiling regex v1.1.0
   Compiling crc32fast v1.1.2
   Compiling ucd-util v0.1.3
   Compiling stable_deref_trait v1.1.1
   Compiling rustc-demangle v0.1.9
   Compiling pub-iterator-type v0.1.1
   Compiling rayon-core v1.4.1
   Compiling unicode-width v0.1.5
   Compiling traitobject v0.0.1
   Compiling pdqselect v0.1.0
   Compiling clamp v0.1.0
   Compiling typeable v0.1.2
   Compiling language-tags v0.2.2
   Compiling linked-hash-map v0.5.1
   Compiling ansi_term v0.11.0
   Compiling itoa v0.1.1
   Compiling quick-error v1.2.2
   Compiling utf8-ranges v1.0.2
   Compiling dtoa v0.2.2
   Compiling odds v0.2.26
   Compiling itoa v0.4.3
   Compiling byteorder v1.2.7
   Compiling crossbeam v0.2.12
   Compiling lazy_static v0.1.16
   Compiling strsim v0.7.0
   Compiling modifier v0.1.0
   Compiling vec_map v0.8.1
   Compiling safemem v0.3.0
   Compiling unicode-segmentation v1.2.1
   Compiling serde v0.8.23
   Compiling rayon v1.0.3
   Compiling regex v0.2.11
   Compiling bitflags v1.0.4
   Compiling take_mut v0.2.2
   Compiling regex-syntax v0.3.9
   Compiling utf8-ranges v0.1.3
   Compiling either v1.5.0
   Compiling termcolor v1.0.4
   Compiling nom v1.2.4
   Compiling byteorder v0.5.3
   Compiling cosmogony v0.3.2
   Compiling dtoa v0.4.3
   Compiling same-file v1.0.4
   Compiling spin v0.4.10
   Compiling fnv v1.0.6
   Compiling maplit v0.1.6
   Compiling git-version v0.2.1
   Compiling quick-error v0.2.2
   Compiling remove_dir_all v0.5.1
   Compiling podio v0.1.6
   Compiling traitobject v0.0.3
   Compiling mdo v0.3.0
   Compiling assert_float_eq v1.1.3
   Compiling log v0.4.6
   Compiling crossbeam-utils v0.2.2
   Compiling unicode-bidi v0.3.4
   Compiling unreachable v1.0.0
   Compiling arrayvec v0.4.8
   Compiling synom v0.11.3
   Compiling unsafe-any v0.4.2
   Compiling phf_shared v0.7.23
   Compiling kernel32-sys v0.2.2
   Compiling rand_core v0.2.2
   Compiling crc v1.8.1
   Compiling unicase v1.4.2
   Compiling memchr v2.1.1
   Compiling thread_local v0.3.6
   Compiling textwrap v0.10.0
   Compiling miniz_oxide v0.2.0
   Compiling error v0.1.9
   Compiling regex-syntax v0.6.4
   Compiling regex-syntax v0.5.6
   Compiling humantime v1.2.0
   Compiling yaml-rust v0.4.2
   Compiling rustc_version v0.1.7
   Compiling itertools v0.7.11
   Compiling openssl-sys v0.7.17
   Compiling base64 v0.9.3
   Compiling term v0.5.1
   Compiling walkdir v2.2.7
   Compiling heck v0.3.0
   Compiling arrayvec v0.3.25
   Compiling smallvec v0.6.7
   Compiling bragi v1.2.0 (/home/agrieco/mimirsbrunn/libs/bragi)
   Compiling mimirsbrunn v1.2.0 (/home/agrieco/mimirsbrunn)
   Compiling typemap v0.3.3
   Compiling phf v0.7.23
   Compiling syn v0.11.11
   Compiling log v0.3.9
   Compiling crossbeam-epoch v0.3.1
   Compiling procinfo v0.3.1
   Compiling openssl-sys-extras v0.7.14
   Compiling openssl v0.7.14
   Compiling plugin v0.2.6
   Compiling hpack v0.2.0
   Compiling mime v0.2.6
error: failed to run custom build command for `openssl v0.7.14`
process didn't exit successfully: `/home/agrieco/mimirsbrunn/target/release/build/openssl-f7ac8fb8e74082c7/build-script-build` (exit code: 101)
--- stdout
TARGET = Some("x86_64-unknown-linux-gnu")
OPT_LEVEL = Some("3")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
CC_x86_64-unknown-linux-gnu = None
CC_x86_64_unknown_linux_gnu = None
HOST_CC = None
CC = None
HOST = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
CFLAGS_x86_64-unknown-linux-gnu = None
CFLAGS_x86_64_unknown_linux_gnu = None
HOST_CFLAGS = None
CFLAGS = None
DEBUG = Some("false")
running: "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "/usr/include/openssl" "-Wall" "-Wextra" "-o" "/home/agrieco/mimirsbrunn/target/release/build/openssl-0c5b4221023529f9/out/src/c_helpers.o" "-c" "src/c_helpers.c"
cargo:warning=src/c_helpers.c: In function 'rust_SSL_clone':
cargo:warning=src/c_helpers.c:4:5: warning: implicit declaration of function 'CRYPTO_add'; did you mean 'CRYPTO_free'? [-Wimplicit-function-declaration]
cargo:warning=     CRYPTO_add(&ssl->references, 1, CRYPTO_LOCK_SSL);
cargo:warning=     ^~~~~~~~~~
cargo:warning=     CRYPTO_free
cargo:warning=src/c_helpers.c:4:20: error: dereferencing pointer to incomplete type 'SSL {aka struct ssl_st}'
cargo:warning=     CRYPTO_add(&ssl->references, 1, CRYPTO_LOCK_SSL);
cargo:warning=                    ^~
cargo:warning=src/c_helpers.c:4:37: error: 'CRYPTO_LOCK_SSL' undeclared (first use in this function); did you mean 'CRYPTO_LOCK'?
cargo:warning=     CRYPTO_add(&ssl->references, 1, CRYPTO_LOCK_SSL);
cargo:warning=                                     ^~~~~~~~~~~~~~~
cargo:warning=                                     CRYPTO_LOCK
cargo:warning=src/c_helpers.c:4:37: note: each undeclared identifier is reported only once for each function it appears in
cargo:warning=src/c_helpers.c: In function 'rust_SSL_CTX_clone':
cargo:warning=src/c_helpers.c:8:20: error: dereferencing pointer to incomplete type 'SSL_CTX {aka struct ssl_ctx_st}'
cargo:warning=     CRYPTO_add(&ctx->references,1,CRYPTO_LOCK_SSL_CTX);
cargo:warning=                    ^~
cargo:warning=src/c_helpers.c:8:35: error: 'CRYPTO_LOCK_SSL_CTX' undeclared (first use in this function); did you mean 'CRYPTO_EX_INDEX_SSL_CTX'?
cargo:warning=     CRYPTO_add(&ctx->references,1,CRYPTO_LOCK_SSL_CTX);
cargo:warning=                                   ^~~~~~~~~~~~~~~~~~~
cargo:warning=                                   CRYPTO_EX_INDEX_SSL_CTX
cargo:warning=src/c_helpers.c: In function 'rust_X509_clone':
cargo:warning=src/c_helpers.c:12:21: error: dereferencing pointer to incomplete type 'X509 {aka struct x509_st}'
cargo:warning=     CRYPTO_add(&x509->references,1,CRYPTO_LOCK_X509);
cargo:warning=                     ^~
cargo:warning=src/c_helpers.c:12:36: error: 'CRYPTO_LOCK_X509' undeclared (first use in this function); did you mean 'CRYPTO_LOCK'?
cargo:warning=     CRYPTO_add(&x509->references,1,CRYPTO_LOCK_X509);
cargo:warning=                                    ^~~~~~~~~~~~~~~~
cargo:warning=                                    CRYPTO_LOCK
exit code: 1

--- stderr
thread 'main' panicked at '

Internal error occurred: Command "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "/usr/include/openssl" "-Wall" "-Wextra" "-o" "/home/agrieco/mimirsbrunn/target/release/build/openssl-0c5b4221023529f9/out/src/c_helpers.o" "-c" "src/c_helpers.c" with args "cc" did not execute successfully (status code exit code: 1).

', /home/agrieco/.cargo/registry/src/github.com-1ecc6299db9ec823/gcc-0.3.55/src/lib.rs:1672:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.

warning: build failed, waiting for other jobs to finish...
error: build failed
```

--- a/README.md
+++ b/README.md
@@ -15,7 +15,8 @@ Mimirsbrunn is composed of several [parts](#components): some of them manage the
 ### Install
 
 - To use the Mimirsbrunn components you need an Elasticsearch database (Elasticsearch version needs to be 2.x).
-- To build you must first install rust following [these instructions](https://www.rust-lang.org/en-US/install.html).
+- To build you must first install rust following [these instructions](https://www.rust-lang.org/tools/install).
+- You need some development packages: make libssl1.0-dev libgeos-dev
 - Then to build Mimirsbrunn:
 ```shell
 cargo build --release

